### PR TITLE
PSVita: touchscreen support

### PR DESCRIPTION
**Pull request by carstene1ns**
This is basically a port of the Nintendo Switch touch UI.
Needs testing on a real vita, since I only tested it with simulated touches on my PSTV.

(Made some small cleanups on the way)

### CODE REVIEW

**Comment by Ghabry** [comment]
cannot really test this, Vita3k crashes (but the continuous build works...)

File: src/platform/psvita/psp2_ui.cpp [hunk 1]
```diff
@@ -71,75 +71,116 @@ namespace {
...
-	
 	for (;;){
-	
-		sceKernelWaitSema(GPU_Mutex, 1, NULL);
-		memcpy(vita2d_texture_get_datap(gpu_texture), vita2d_texture_get_datap(next_texture), vita2d_texture_get_stride(gpu_texture)*240);
+
+		sceKernelWaitSema(GPU_Mutex, 1, nullptr);
+		memcpy(vita2d_texture_get_datap(gpu_texture),
+			vita2d_texture_get_datap(next_texture),
+			vita2d_texture_get_stride(gpu_texture)*240);
```
Comment (Ghabry) [inline src/platform/psvita/psp2_ui.cpp:109]: wrong indent

File: src/platform/psvita/psp2_ui.cpp [hunk 2]
```diff
@@ -217,50 +276,51 @@ void Psp2Ui::ProcessEvents() {
...
+	}
 }
 
 void Psp2Ui::UpdateDisplay() {
-	sceKernelWaitSema(GPU_Mutex, 1, NULL);
-	memcpy(vita2d_texture_get_datap(next_texture), vita2d_texture_get_datap(main_texture), vita2d_texture_get_stride(main_texture)*240);
+	sceKernelWaitSema(GPU_Mutex, 1, nullptr);
+	memcpy(vita2d_texture_get_datap(next_texture),
+		vita2d_texture_get_datap(main_texture),
+		vita2d_texture_get_stride(main_texture)*240);
```
Comment (Ghabry) [inline src/platform/psvita/psp2_ui.cpp:323]: same

File: src/platform/psvita/psp2_ui.cpp (hunk 1 above)
Comment (carstene1ns) [reply src/platform/psvita/psp2_ui.cpp:109]: I indented all function arguments spanning over multiple lines with one additional tab, we really need a policy for this (coding style).

Comment (Ghabry) [reply src/platform/psvita/psp2_ui.cpp:109]: oh sorry I did not see this is part of the argument list :facepalm:

**Review by fdelapena — APPROVED** [review]

**Review by Ghabry — APPROVED** [review]
Not testable by me, code looks fine.

@carstene1ns Can I have a screenshot or photo for the blog post? (with visible touch ui) :)

**Comment by carstene1ns** [comment]
Merging this now, to maybe get some feedback.

**Comment by ttyborg** [comment]
Hi!

Just installed 0.7 on my Vita 1000, and it seems the touch area is misaligned on the vertical axis.
I mean, the top of the area is fine, but if I want to touch the bottom row buttons, I have to touch the middle of the screen.

**Comment by Ghabry** [comment]
Thanks created a new bug for this https://github.com/EasyRPG/Player/issues/2683
@ttyborg

**Pull request merged**